src/ui/shared/Checkbox.tsx

```diff
@@ -0,0 +1,89 @@
+// eslint-disable-next-line no-use-before-define
+import React from 'react'
+import styled from 'styled-components'
+import { CHECKBOX_CLASSNAME } from '../../constants/cssSelectors'
+
+const CheckboxContainer = styled.div`
+  display: inline-block;
+  vertical-align: middle;
+`
+
+const Icon = styled.svg`
+  fill: none;
+  stroke: white;
+  stroke-width: 2px;
+`
+// Hide checkbox visually but remain accessible to screen readers.
+// Source: https://polished.js.org/docs/#hidevisually
+const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
+  border: 0;
+  clip: rect(0 0 0 0);
+  clip-path: inset(50%);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+`
+
+interface StyledCheckboxProps {
+    checked: boolean;
+}
+
+const StyledCheckbox = styled.div<StyledCheckboxProps>`
+  display: inline-block;
+  width: 16px;
+  height: 16px;
+  background: ${props => (props.checked ? props.theme.primaryColor : props.theme.p)};
+  border-radius: 3px;
+  transition: all 150ms;
+
+  ${HiddenCheckbox}:focus + & {
+    box-shadow: ${props => `0 0 0 3px ${shadeColor(props.theme.primaryColor, 20)}`};
+  }
+
+  ${Icon} {
+    visibility: ${props => (props.checked ? 'visible' : 'hidden')}
+  }
+`
+
+interface CheckboxInterface {
+    className?: string,
+    checked: boolean,
+    onChange?: () => void
+}
+
+const Checkbox: React.FC<CheckboxInterface> = ({ className, checked, onChange }) => (
+  <CheckboxContainer className={className}>
+    <HiddenCheckbox checked={checked} className={CHECKBOX_CLASSNAME} onChange={onChange} />
+    <StyledCheckbox checked={checked}>
+      <Icon viewBox="0 0 24 24">
+        <polyline points="20 6 9 17 4 12" />
+      </Icon>
+    </StyledCheckbox>
+  </CheckboxContainer>
+)
+
+export default Checkbox
+
+function shadeColor (color: string, percent: number) {
+  let R = parseInt(color.substring(1, 3), 16)
+  let G = parseInt(color.substring(3, 5), 16)
+  let B = parseInt(color.substring(5, 7), 16)
+
+  R = Math.floor(R * (100 + percent) / 100)
+  G = Math.floor(G * (100 + percent) / 100)
+  B = Math.floor(B * (100 + percent) / 100)
+
+  R = (R < 255) ? R : 255
+  G = (G < 255) ? G : 255
+  B = (B < 255) ? B : 255
+
+  const RR = ((R.toString(16).length === 1) ? '0' + R.toString(16) : R.toString(16))
+  const GG = ((G.toString(16).length === 1) ? '0' + G.toString(16) : G.toString(16))
+  const BB = ((B.toString(16).length === 1) ? '0' + B.toString(16) : B.toString(16))
+
+  return '#' + RR + GG + BB
+}
```

Comment:
I don't fully follow what this code is doing for the drop-shadow color, but can this be done with CSS instead?

Reply:
This code is doing something similar to the css rgba function, the idea is to use any theme color and lighten it. I couldn find a way to do it in css because i wanted a lighten color for a shadow box. Alternatively, we could add an extra color in the theme. Let me know what you think.

Reply:
Could you use something like:

```
.whatever {
     filter: brightness(120%)
}
```

to lighten it up? If not, the JS is fine, it just feels like extra bloat in the repo.

Reply:
shadow box doesn't work with filter, but I found an easier way so I removed the shadeColor function.